
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://felixkrueger.github.io/SNPsplit/Examples/examples/">
      
      
        <link rel="prev" href="../../SNPsplit/specific_comments/">
      
      
        <link rel="next" href="../../options/genome_prep_options/">
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.0.2">
    
    
      
        <title>Examples - SNPsplit</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.f56500e0.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.2505c338.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Noto+Sans:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Noto Sans";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
      
        <meta  property="og:type"  content="website" >
      
        <meta  property="og:title"  content="Examples - SNPsplit" >
      
        <meta  property="og:description"  content="None" >
      
        <meta  property="og:image"  content="https://felixkrueger.github.io/SNPsplit/assets/images/social/Examples/examples.png" >
      
        <meta  property="og:image:type"  content="image/png" >
      
        <meta  property="og:image:width"  content="1200" >
      
        <meta  property="og:image:height"  content="630" >
      
        <meta  property="og:url"  content="https://felixkrueger.github.io/SNPsplit/Examples/examples/" >
      
        <meta  name="twitter:card"  content="summary_large_image" >
      
        <meta  name="twitter:title"  content="Examples - SNPsplit" >
      
        <meta  name="twitter:description"  content="None" >
      
        <meta  name="twitter:image"  content="https://felixkrueger.github.io/SNPsplit/assets/images/social/Examples/examples.png" >
      
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="slate" data-md-color-primary="teal" data-md-color-accent="blue">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#paired-end-report-2x50bp" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="SNPsplit" class="md-header__button md-logo" aria-label="SNPsplit" data-md-component="logo">
      
  <img src="../../images/SNPsplit.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            SNPsplit
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Examples
            
          </span>
        </div>
      </div>
    </div>
    
      <form class="md-header__option" data-md-component="palette">
        
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="teal" data-md-color-accent="blue"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
          
            <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_2" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3 3.19.09m3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95 2.06.05m-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31Z"/></svg>
            </label>
          
        
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="blue"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_2">
          
            <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0-7 2.39 3.42C13.65 5.15 12.84 5 12 5c-.84 0-1.65.15-2.39.42L12 2M3.34 7l4.16-.35A7.2 7.2 0 0 0 5.94 8.5c-.44.74-.69 1.5-.83 2.29L3.34 7m.02 10 1.76-3.77a7.131 7.131 0 0 0 2.38 4.14L3.36 17M20.65 7l-1.77 3.79a7.023 7.023 0 0 0-2.38-4.15l4.15.36m-.01 10-4.14.36c.59-.51 1.12-1.14 1.54-1.86.42-.73.69-1.5.83-2.29L20.64 17M12 22l-2.41-3.44c.74.27 1.55.44 2.41.44.82 0 1.63-.17 2.37-.44L12 22Z"/></svg>
            </label>
          
        
      </form>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/FelixKrueger/SNPsplit" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    FelixKrueger/SNPsplit
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  <li class="md-tabs__item">
    <a href="../.." class="md-tabs__link">
      Overview
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../workflow/" class="md-tabs__link">
      SNPsplit workflow
    </a>
  </li>

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../genome_prep/genome_preparation/" class="md-tabs__link">
        Genome preparation
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../SNPsplit/details/" class="md-tabs__link">
        SNPsplit
      </a>
    </li>
  

      
        
  
  
    
  


  
  
  
    <li class="md-tabs__item">
      <a href="./" class="md-tabs__link md-tabs__link--active">
        Examples
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../options/genome_prep_options/" class="md-tabs__link">
        Full list of options
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
                
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" hidden>
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="SNPsplit" class="md-nav__button md-logo" aria-label="SNPsplit" data-md-component="logo">
      
  <img src="../../images/SNPsplit.png" alt="logo">

    </a>
    SNPsplit
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/FelixKrueger/SNPsplit" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    FelixKrueger/SNPsplit
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Overview
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../workflow/" class="md-nav__link">
        SNPsplit workflow
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" >
      
      
      
      
        <label class="md-nav__link" for="__nav_3">
          Genome preparation
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Genome preparation" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Genome preparation
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../genome_prep/genome_preparation/" class="md-nav__link">
        SNPsplit genome preparation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../genome_prep/legacy/" class="md-nav__link">
        Legacy (v5/GRCm38)
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" >
      
      
      
      
        <label class="md-nav__link" for="__nav_4">
          SNPsplit
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="SNPsplit" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          SNPsplit
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../SNPsplit/details/" class="md-nav__link">
        The SNPsplit workflow in more detail
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../SNPsplit/specific_comments/" class="md-nav__link">
        Specific comments
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Examples
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Examples" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Examples
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Examples
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Examples
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#paired-end-report-2x50bp" class="md-nav__link">
    Paired-end report (2x50bp):
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#hi-c-report-2x100bp" class="md-nav__link">
    Hi-C report (2x100bp):
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bs-seq-report-2x100bp" class="md-nav__link">
    BS-Seq report (2x100bp):
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6" type="checkbox" id="__nav_6" >
      
      
      
      
        <label class="md-nav__link" for="__nav_6">
          Full list of options
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Full list of options" data-md-level="1">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Full list of options
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../options/genome_prep_options/" class="md-nav__link">
        Genome preparation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../options/SNPsplit_options/" class="md-nav__link">
        SNPsplit
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
                
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#paired-end-report-2x50bp" class="md-nav__link">
    Paired-end report (2x50bp):
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#hi-c-report-2x100bp" class="md-nav__link">
    Hi-C report (2x100bp):
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#bs-seq-report-2x100bp" class="md-nav__link">
    BS-Seq report (2x100bp):
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  



  <h1>Examples</h1>

<h2 id="paired-end-report-2x50bp">Paired-end report (2x50bp):</h2>
<p><div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>Input file:                 &#39;FVBNJ_Cast.bam&#39;
<a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>Writing allele-flagged output file to:  &#39;FVBNJ_Cast.allele_flagged.bam&#39;
<a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>
<a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a>
<a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a>Allele-tagging report
<a id="__codelineno-0-6" name="__codelineno-0-6" href="#__codelineno-0-6"></a>=====================
<a id="__codelineno-0-7" name="__codelineno-0-7" href="#__codelineno-0-7"></a>Processed 194564995 read alignments in total
<a id="__codelineno-0-8" name="__codelineno-0-8" href="#__codelineno-0-8"></a>149380724 reads were unassignable (76.78%)
<a id="__codelineno-0-9" name="__codelineno-0-9" href="#__codelineno-0-9"></a>35143075 reads were specific for genome 1 (18.06%)
<a id="__codelineno-0-10" name="__codelineno-0-10" href="#__codelineno-0-10"></a>9860248 reads were specific for genome 2 (5.07%)
<a id="__codelineno-0-11" name="__codelineno-0-11" href="#__codelineno-0-11"></a>118662 reads did not contain one of the expected bases at known SNP positions (0.06%)
<a id="__codelineno-0-12" name="__codelineno-0-12" href="#__codelineno-0-12"></a>180948 contained conflicting allele-specific SNPs (0.09%)
<a id="__codelineno-0-13" name="__codelineno-0-13" href="#__codelineno-0-13"></a>
<a id="__codelineno-0-14" name="__codelineno-0-14" href="#__codelineno-0-14"></a>
<a id="__codelineno-0-15" name="__codelineno-0-15" href="#__codelineno-0-15"></a>SNP coverage report
<a id="__codelineno-0-16" name="__codelineno-0-16" href="#__codelineno-0-16"></a>===================
<a id="__codelineno-0-17" name="__codelineno-0-17" href="#__codelineno-0-17"></a>N-containing reads: 45276050
<a id="__codelineno-0-18" name="__codelineno-0-18" href="#__codelineno-0-18"></a>non-N:          149262062
<a id="__codelineno-0-19" name="__codelineno-0-19" href="#__codelineno-0-19"></a>total:          194564995
<a id="__codelineno-0-20" name="__codelineno-0-20" href="#__codelineno-0-20"></a>Reads had a deletion of the N-masked position (and were thus dropped):  26883 (0.01%)
<a id="__codelineno-0-21" name="__codelineno-0-21" href="#__codelineno-0-21"></a>Of which had multiple deletions of N-masked positions within the same read: 30
<a id="__codelineno-0-22" name="__codelineno-0-22" href="#__codelineno-0-22"></a>
<a id="__codelineno-0-23" name="__codelineno-0-23" href="#__codelineno-0-23"></a>Of valid N containing reads,
<a id="__codelineno-0-24" name="__codelineno-0-24" href="#__codelineno-0-24"></a>N was present in the list of known SNPs:        61087551 (99.99%)
<a id="__codelineno-0-25" name="__codelineno-0-25" href="#__codelineno-0-25"></a>N was not present in the list of SNPs:      4773 (0.01%)
<a id="__codelineno-0-26" name="__codelineno-0-26" href="#__codelineno-0-26"></a>
<a id="__codelineno-0-27" name="__codelineno-0-27" href="#__codelineno-0-27"></a>Input file:                     &#39;FVBNJ_Cast.allele_flagged.bam&#39;
<a id="__codelineno-0-28" name="__codelineno-0-28" href="#__codelineno-0-28"></a>Writing unassigned reads to:                &#39;FVBNJ_Cast.unassigned.bam&#39;
<a id="__codelineno-0-29" name="__codelineno-0-29" href="#__codelineno-0-29"></a>Writing genome 1-specific reads to:         &#39;FVBNJ_Cast.genome1.bam&#39;
<a id="__codelineno-0-30" name="__codelineno-0-30" href="#__codelineno-0-30"></a>Writing genome 2-specific reads to:         &#39;FVBNJ_Cast.genome2.bam&#39;
<a id="__codelineno-0-31" name="__codelineno-0-31" href="#__codelineno-0-31"></a>
<a id="__codelineno-0-32" name="__codelineno-0-32" href="#__codelineno-0-32"></a>
<a id="__codelineno-0-33" name="__codelineno-0-33" href="#__codelineno-0-33"></a>Allele-specific paired-end sorting report
<a id="__codelineno-0-34" name="__codelineno-0-34" href="#__codelineno-0-34"></a>=========================================
<a id="__codelineno-0-35" name="__codelineno-0-35" href="#__codelineno-0-35"></a>Read pairs/singletons processed in total:       98215744
<a id="__codelineno-0-36" name="__codelineno-0-36" href="#__codelineno-0-36"></a>    thereof were read pairs:            96349251
<a id="__codelineno-0-37" name="__codelineno-0-37" href="#__codelineno-0-37"></a>    thereof were singletons:            1866493
<a id="__codelineno-0-38" name="__codelineno-0-38" href="#__codelineno-0-38"></a>Reads were unassignable (not overlapping SNPs): 61174812 (62.29%)
<a id="__codelineno-0-39" name="__codelineno-0-39" href="#__codelineno-0-39"></a>    thereof were read pairs:    59662537
<a id="__codelineno-0-40" name="__codelineno-0-40" href="#__codelineno-0-40"></a>    thereof were singletons:    1512275
<a id="__codelineno-0-41" name="__codelineno-0-41" href="#__codelineno-0-41"></a>Reads were specific for genome 1:           28657857 (29.18%)
<a id="__codelineno-0-42" name="__codelineno-0-42" href="#__codelineno-0-42"></a>    thereof were read pairs:    28446094
<a id="__codelineno-0-43" name="__codelineno-0-43" href="#__codelineno-0-43"></a>    thereof were singletons:    211763
<a id="__codelineno-0-44" name="__codelineno-0-44" href="#__codelineno-0-44"></a>Reads were specific for genome 2:           8122687 (8.27%)
<a id="__codelineno-0-45" name="__codelineno-0-45" href="#__codelineno-0-45"></a>    thereof were read pairs:    7985424
<a id="__codelineno-0-46" name="__codelineno-0-46" href="#__codelineno-0-46"></a>    thereof were singletons:    137263
<a id="__codelineno-0-47" name="__codelineno-0-47" href="#__codelineno-0-47"></a>Reads contained conflicting SNP information:        260388 (0.27%)
<a id="__codelineno-0-48" name="__codelineno-0-48" href="#__codelineno-0-48"></a>    thereof were read pairs:    255196
<a id="__codelineno-0-49" name="__codelineno-0-49" href="#__codelineno-0-49"></a>    thereof were singletons:    5192
</code></pre></div>
â€ƒ</p>
<h2 id="hi-c-report-2x100bp">Hi-C report (2x100bp):</h2>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>Input file:                 Black6_129S1.bam
<a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>Writing allele-flagged output file to:  Black6_129S1.allele_flagged.bam
<a id="__codelineno-1-3" name="__codelineno-1-3" href="#__codelineno-1-3"></a>
<a id="__codelineno-1-4" name="__codelineno-1-4" href="#__codelineno-1-4"></a>
<a id="__codelineno-1-5" name="__codelineno-1-5" href="#__codelineno-1-5"></a>Allele-tagging report
<a id="__codelineno-1-6" name="__codelineno-1-6" href="#__codelineno-1-6"></a>=====================
<a id="__codelineno-1-7" name="__codelineno-1-7" href="#__codelineno-1-7"></a>Processed 94887256 read alignments in total
<a id="__codelineno-1-8" name="__codelineno-1-8" href="#__codelineno-1-8"></a>59662038 reads were unassignable (62.88%)
<a id="__codelineno-1-9" name="__codelineno-1-9" href="#__codelineno-1-9"></a>19851697 reads were specific for genome 1 (20.92%)
<a id="__codelineno-1-10" name="__codelineno-1-10" href="#__codelineno-1-10"></a>15047281 reads were specific for genome 2 (15.86%)
<a id="__codelineno-1-11" name="__codelineno-1-11" href="#__codelineno-1-11"></a>47261 reads did not contain one of the expected bases at known SNP positions (0.05%)
<a id="__codelineno-1-12" name="__codelineno-1-12" href="#__codelineno-1-12"></a>326240 contained conflicting allele-specific SNPs (0.34%)
<a id="__codelineno-1-13" name="__codelineno-1-13" href="#__codelineno-1-13"></a>
<a id="__codelineno-1-14" name="__codelineno-1-14" href="#__codelineno-1-14"></a>
<a id="__codelineno-1-15" name="__codelineno-1-15" href="#__codelineno-1-15"></a>SNP coverage report
<a id="__codelineno-1-16" name="__codelineno-1-16" href="#__codelineno-1-16"></a>===================
<a id="__codelineno-1-17" name="__codelineno-1-17" href="#__codelineno-1-17"></a>N-containing reads: 35231977
<a id="__codelineno-1-18" name="__codelineno-1-18" href="#__codelineno-1-18"></a>non-N:          59614777
<a id="__codelineno-1-19" name="__codelineno-1-19" href="#__codelineno-1-19"></a>total:          94887256
<a id="__codelineno-1-20" name="__codelineno-1-20" href="#__codelineno-1-20"></a>Reads had a deletion of the N-masked position (and were thus dropped):  40502 (0.04%)
<a id="__codelineno-1-21" name="__codelineno-1-21" href="#__codelineno-1-21"></a>Of which had multiple deletions of N-masked positions within the same read: 59
<a id="__codelineno-1-22" name="__codelineno-1-22" href="#__codelineno-1-22"></a>
<a id="__codelineno-1-23" name="__codelineno-1-23" href="#__codelineno-1-23"></a>Of valid N containing reads,
<a id="__codelineno-1-24" name="__codelineno-1-24" href="#__codelineno-1-24"></a>N was present in the list of known SNPs:    57101748 (99.99%)
<a id="__codelineno-1-25" name="__codelineno-1-25" href="#__codelineno-1-25"></a>N was not present in the list of SNPs:  4211 (0.01%)
<a id="__codelineno-1-26" name="__codelineno-1-26" href="#__codelineno-1-26"></a>
<a id="__codelineno-1-27" name="__codelineno-1-27" href="#__codelineno-1-27"></a>Input file:                     Black6_129S1.allele_flagged.bam&#39;
<a id="__codelineno-1-28" name="__codelineno-1-28" href="#__codelineno-1-28"></a>Writing unassigned reads to:                Black6_129S1.UA_UA.bam&#39;
<a id="__codelineno-1-29" name="__codelineno-1-29" href="#__codelineno-1-29"></a>Writing genome 1-specific reads to:         Black6_129S1.G1_G1.bam&#39;
<a id="__codelineno-1-30" name="__codelineno-1-30" href="#__codelineno-1-30"></a>Writing genome 2-specific reads to:         Black6_129S1.G2_G2.bam&#39;
<a id="__codelineno-1-31" name="__codelineno-1-31" href="#__codelineno-1-31"></a>Writing G1/UA reads to:             Black6_129S1.G1_UA.bam&#39;
<a id="__codelineno-1-32" name="__codelineno-1-32" href="#__codelineno-1-32"></a>Writing G2/UA reads to:             Black6_129S1.G2_UA.bam&#39;
<a id="__codelineno-1-33" name="__codelineno-1-33" href="#__codelineno-1-33"></a>Writing G1/G2 reads to:             Black6_129S1.G1_G2.bam&#39;
<a id="__codelineno-1-34" name="__codelineno-1-34" href="#__codelineno-1-34"></a>
<a id="__codelineno-1-35" name="__codelineno-1-35" href="#__codelineno-1-35"></a>
<a id="__codelineno-1-36" name="__codelineno-1-36" href="#__codelineno-1-36"></a>Allele-specific paired-end sorting report
<a id="__codelineno-1-37" name="__codelineno-1-37" href="#__codelineno-1-37"></a>=========================================
<a id="__codelineno-1-38" name="__codelineno-1-38" href="#__codelineno-1-38"></a>Read pairs processed in total:          47443628
<a id="__codelineno-1-39" name="__codelineno-1-39" href="#__codelineno-1-39"></a>Read pairs were unassignable (UA/UA):           18862725 (39.76%)
<a id="__codelineno-1-40" name="__codelineno-1-40" href="#__codelineno-1-40"></a>Read pairs were specific for genome 1 (G1/G1):  3533932 (7.45%)
<a id="__codelineno-1-41" name="__codelineno-1-41" href="#__codelineno-1-41"></a>Read pairs were specific for genome 2 (G2/G2):  2592040 (5.46%)
<a id="__codelineno-1-42" name="__codelineno-1-42" href="#__codelineno-1-42"></a>Read pairs were a mix of G1 and UA:         12306421 (25.94%). Of these,
<a id="__codelineno-1-43" name="__codelineno-1-43" href="#__codelineno-1-43"></a>            were G1/UA: 6018598
<a id="__codelineno-1-44" name="__codelineno-1-44" href="#__codelineno-1-44"></a>            were UA/G1: 6287823
<a id="__codelineno-1-45" name="__codelineno-1-45" href="#__codelineno-1-45"></a>Read pairs were a mix of G2 and UA:         9430675 (19.88%). Of these,
<a id="__codelineno-1-46" name="__codelineno-1-46" href="#__codelineno-1-46"></a>            were G2/UA: 4603429
<a id="__codelineno-1-47" name="__codelineno-1-47" href="#__codelineno-1-47"></a>            were UA/G2: 4827246
<a id="__codelineno-1-48" name="__codelineno-1-48" href="#__codelineno-1-48"></a>Read pairs were a mix of G1 and G2:         395296 (0.83%). Of these,
<a id="__codelineno-1-49" name="__codelineno-1-49" href="#__codelineno-1-49"></a>            were G1/G2: 198330
<a id="__codelineno-1-50" name="__codelineno-1-50" href="#__codelineno-1-50"></a>            were G2/G1: 196966
<a id="__codelineno-1-51" name="__codelineno-1-51" href="#__codelineno-1-51"></a>Read pairs contained conflicting SNP information:   322539 (0.68%)
</code></pre></div>
<h2 id="bs-seq-report-2x100bp">BS-Seq report (2x100bp):</h2>
<div class="highlight"><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>Input file:                 &#39;129_Cast_bismark_bt2_pe.bam&#39;
<a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>Writing allele-flagged output file to:  &#39;129_Cast_bismark_bt2_pe.allele_flagged.bam&#39;
<a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a>
<a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a>
<a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a>Allele-tagging report
<a id="__codelineno-2-6" name="__codelineno-2-6" href="#__codelineno-2-6"></a>=====================
<a id="__codelineno-2-7" name="__codelineno-2-7" href="#__codelineno-2-7"></a>Processed 162441396 read alignments in total
<a id="__codelineno-2-8" name="__codelineno-2-8" href="#__codelineno-2-8"></a>Reads were unaligned and hence skipped: 0 (0.00%)
<a id="__codelineno-2-9" name="__codelineno-2-9" href="#__codelineno-2-9"></a>109109113 reads were unassignable (67.17%)
<a id="__codelineno-2-10" name="__codelineno-2-10" href="#__codelineno-2-10"></a>30267901 reads were specific for genome 1 (18.63%)
<a id="__codelineno-2-11" name="__codelineno-2-11" href="#__codelineno-2-11"></a>22697499 reads were specific for genome 2 (13.97%)
<a id="__codelineno-2-12" name="__codelineno-2-12" href="#__codelineno-2-12"></a>15807753 reads did not contain one of the expected bases at known SNP positions (9.73%)
<a id="__codelineno-2-13" name="__codelineno-2-13" href="#__codelineno-2-13"></a>366883 contained conflicting allele-specific SNPs (0.23%)
<a id="__codelineno-2-14" name="__codelineno-2-14" href="#__codelineno-2-14"></a>
<a id="__codelineno-2-15" name="__codelineno-2-15" href="#__codelineno-2-15"></a>
<a id="__codelineno-2-16" name="__codelineno-2-16" href="#__codelineno-2-16"></a>SNP coverage report
<a id="__codelineno-2-17" name="__codelineno-2-17" href="#__codelineno-2-17"></a>===================
<a id="__codelineno-2-18" name="__codelineno-2-18" href="#__codelineno-2-18"></a>SNP annotation file:    ../all_Cast_SNPs_129S1_reference.mgp.v4.txt.gz
<a id="__codelineno-2-19" name="__codelineno-2-19" href="#__codelineno-2-19"></a>N-containing reads: 68984287
<a id="__codelineno-2-20" name="__codelineno-2-20" href="#__codelineno-2-20"></a>non-N:          93301360
<a id="__codelineno-2-21" name="__codelineno-2-21" href="#__codelineno-2-21"></a>total:          162441396
<a id="__codelineno-2-22" name="__codelineno-2-22" href="#__codelineno-2-22"></a>Reads had a deletion of the N-masked position (and were thus dropped):  155749 (0.10%)
<a id="__codelineno-2-23" name="__codelineno-2-23" href="#__codelineno-2-23"></a>Of which had multiple deletions of N-masked positions within the same read: 65
<a id="__codelineno-2-24" name="__codelineno-2-24" href="#__codelineno-2-24"></a>
<a id="__codelineno-2-25" name="__codelineno-2-25" href="#__codelineno-2-25"></a>Of valid N containing reads,
<a id="__codelineno-2-26" name="__codelineno-2-26" href="#__codelineno-2-26"></a>N was present in the list of known SNPs:    119119643 (99.99%)
<a id="__codelineno-2-27" name="__codelineno-2-27" href="#__codelineno-2-27"></a>Positions were skipped since they involved C&gt;T SNPs:    38464451
<a id="__codelineno-2-28" name="__codelineno-2-28" href="#__codelineno-2-28"></a>N was not present in the list of SNPs:      7517 (0.01%)
<a id="__codelineno-2-29" name="__codelineno-2-29" href="#__codelineno-2-29"></a>
<a id="__codelineno-2-30" name="__codelineno-2-30" href="#__codelineno-2-30"></a>Input file:                     129_Cast_bismark_bt2_pe.allele_flagged.bam&#39;
<a id="__codelineno-2-31" name="__codelineno-2-31" href="#__codelineno-2-31"></a>Writing unassigned reads to:                129_Cast_bismark_bt2_pe.unassigned.bam&#39;
<a id="__codelineno-2-32" name="__codelineno-2-32" href="#__codelineno-2-32"></a>Writing genome 1-specific reads to:         129_Cast_bismark_bt2_pe.genome1.bam&#39;
<a id="__codelineno-2-33" name="__codelineno-2-33" href="#__codelineno-2-33"></a>Writing genome 2-specific reads to:         129_Cast_bismark_bt2_pe.genome2.bam&#39;
<a id="__codelineno-2-34" name="__codelineno-2-34" href="#__codelineno-2-34"></a>
<a id="__codelineno-2-35" name="__codelineno-2-35" href="#__codelineno-2-35"></a>
<a id="__codelineno-2-36" name="__codelineno-2-36" href="#__codelineno-2-36"></a>Allele-specific paired-end sorting report
<a id="__codelineno-2-37" name="__codelineno-2-37" href="#__codelineno-2-37"></a>=========================================
<a id="__codelineno-2-38" name="__codelineno-2-38" href="#__codelineno-2-38"></a>Read pairs/singletons processed in total:       81220698
<a id="__codelineno-2-39" name="__codelineno-2-39" href="#__codelineno-2-39"></a>    thereof were read pairs:            81220698
<a id="__codelineno-2-40" name="__codelineno-2-40" href="#__codelineno-2-40"></a>    thereof were singletons:            0
<a id="__codelineno-2-41" name="__codelineno-2-41" href="#__codelineno-2-41"></a>Reads were unassignable (not overlapping SNPs):     40420625 (49.77%)
<a id="__codelineno-2-42" name="__codelineno-2-42" href="#__codelineno-2-42"></a>    thereof were read pairs:    40420625
<a id="__codelineno-2-43" name="__codelineno-2-43" href="#__codelineno-2-43"></a>    thereof were singletons:    0
<a id="__codelineno-2-44" name="__codelineno-2-44" href="#__codelineno-2-44"></a>Reads were specific for genome 1:           23037433 (28.36%)
<a id="__codelineno-2-45" name="__codelineno-2-45" href="#__codelineno-2-45"></a>    thereof were read pairs:    23037433
<a id="__codelineno-2-46" name="__codelineno-2-46" href="#__codelineno-2-46"></a>    thereof were singletons:    0
<a id="__codelineno-2-47" name="__codelineno-2-47" href="#__codelineno-2-47"></a>Reads were specific for genome 2:           17303663 (21.30%)
<a id="__codelineno-2-48" name="__codelineno-2-48" href="#__codelineno-2-48"></a>    thereof were read pairs:    17303663
<a id="__codelineno-2-49" name="__codelineno-2-49" href="#__codelineno-2-49"></a>    thereof were singletons:    0
<a id="__codelineno-2-50" name="__codelineno-2-50" href="#__codelineno-2-50"></a>Reads contained conflicting SNP information:        458977 (0.57%)
<a id="__codelineno-2-51" name="__codelineno-2-51" href="#__codelineno-2-51"></a>    thereof were read pairs:    458977
<a id="__codelineno-2-52" name="__codelineno-2-52" href="#__codelineno-2-52"></a>    thereof were singletons:    0
</code></pre></div>



  



                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.instant", "navigation.tabs"], "search": "../../assets/javascripts/workers/search.12658920.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.5cf534bf.min.js"></script>
      
    
  </body>
</html>