
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="https://felixkrueger.github.io/SNPsplit/genome_prep/genome_preparation/">
      
      
        <link rel="prev" href="../../workflow/">
      
      
        <link rel="next" href="../legacy/">
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.4.2, mkdocs-material-9.0.2">
    
    
      
        <title>SNPsplit genome preparation - SNPsplit</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.f56500e0.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.2505c338.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Noto+Sans:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Noto Sans";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
      
        <meta  property="og:type"  content="website" >
      
        <meta  property="og:title"  content="SNPsplit genome preparation - SNPsplit" >
      
        <meta  property="og:description"  content="None" >
      
        <meta  property="og:image"  content="https://felixkrueger.github.io/SNPsplit/assets/images/social/genome_prep/genome_preparation.png" >
      
        <meta  property="og:image:type"  content="image/png" >
      
        <meta  property="og:image:width"  content="1200" >
      
        <meta  property="og:image:height"  content="630" >
      
        <meta  property="og:url"  content="https://felixkrueger.github.io/SNPsplit/genome_prep/genome_preparation/" >
      
        <meta  name="twitter:card"  content="summary_large_image" >
      
        <meta  name="twitter:title"  content="SNPsplit genome preparation - SNPsplit" >
      
        <meta  name="twitter:description"  content="None" >
      
        <meta  name="twitter:image"  content="https://felixkrueger.github.io/SNPsplit/assets/images/social/genome_prep/genome_preparation.png" >
      
    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="slate" data-md-color-primary="teal" data-md-color-accent="blue">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#snpsplit-genome-preparation" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="SNPsplit" class="md-header__button md-logo" aria-label="SNPsplit" data-md-component="logo">
      
  <img src="../../images/SNPsplit.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            SNPsplit
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              SNPsplit genome preparation
            
          </span>
        </div>
      </div>
    </div>
    
      <form class="md-header__option" data-md-component="palette">
        
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="teal" data-md-color-accent="blue"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
          
            <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_2" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m17.75 4.09-2.53 1.94.91 3.06-2.63-1.81-2.63 1.81.91-3.06-2.53-1.94L12.44 4l1.06-3 1.06 3 3.19.09m3.5 6.91-1.64 1.25.59 1.98-1.7-1.17-1.7 1.17.59-1.98L15.75 11l2.06-.05L18.5 9l.69 1.95 2.06.05m-2.28 4.95c.83-.08 1.72 1.1 1.19 1.85-.32.45-.66.87-1.08 1.27C15.17 23 8.84 23 4.94 19.07c-3.91-3.9-3.91-10.24 0-14.14.4-.4.82-.76 1.27-1.08.75-.53 1.93.36 1.85 1.19-.27 2.86.69 5.83 2.89 8.02a9.96 9.96 0 0 0 8.02 2.89m-1.64 2.02a12.08 12.08 0 0 1-7.8-3.47c-2.17-2.19-3.33-5-3.49-7.82-2.81 3.14-2.7 7.96.31 10.98 3.02 3.01 7.84 3.12 10.98.31Z"/></svg>
            </label>
          
        
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="teal" data-md-color-accent="blue"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_2">
          
            <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 7a5 5 0 0 1 5 5 5 5 0 0 1-5 5 5 5 0 0 1-5-5 5 5 0 0 1 5-5m0 2a3 3 0 0 0-3 3 3 3 0 0 0 3 3 3 3 0 0 0 3-3 3 3 0 0 0-3-3m0-7 2.39 3.42C13.65 5.15 12.84 5 12 5c-.84 0-1.65.15-2.39.42L12 2M3.34 7l4.16-.35A7.2 7.2 0 0 0 5.94 8.5c-.44.74-.69 1.5-.83 2.29L3.34 7m.02 10 1.76-3.77a7.131 7.131 0 0 0 2.38 4.14L3.36 17M20.65 7l-1.77 3.79a7.023 7.023 0 0 0-2.38-4.15l4.15.36m-.01 10-4.14.36c.59-.51 1.12-1.14 1.54-1.86.42-.73.69-1.5.83-2.29L20.64 17M12 22l-2.41-3.44c.74.27 1.55.44 2.41.44.82 0 1.63-.17 2.37-.44L12 22Z"/></svg>
            </label>
          
        
      </form>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/FelixKrueger/SNPsplit" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    FelixKrueger/SNPsplit
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
            
<nav class="md-tabs" aria-label="Tabs" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  
  


  <li class="md-tabs__item">
    <a href="../.." class="md-tabs__link">
      Overview
    </a>
  </li>

      
        
  
  


  <li class="md-tabs__item">
    <a href="../../workflow/" class="md-tabs__link">
      SNPsplit workflow
    </a>
  </li>

      
        
  
  
    
  


  
  
  
    <li class="md-tabs__item">
      <a href="./" class="md-tabs__link md-tabs__link--active">
        Genome preparation
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../SNPsplit/details/" class="md-tabs__link">
        SNPsplit
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../Examples/examples/" class="md-tabs__link">
        Examples
      </a>
    </li>
  

      
        
  
  


  
  
  
    <li class="md-tabs__item">
      <a href="../../options/genome_prep_options/" class="md-tabs__link">
        Full list of options
      </a>
    </li>
  

      
    </ul>
  </div>
</nav>
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

  


<nav class="md-nav md-nav--primary md-nav--lifted" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="SNPsplit" class="md-nav__button md-logo" aria-label="SNPsplit" data-md-component="logo">
      
  <img src="../../images/SNPsplit.png" alt="logo">

    </a>
    SNPsplit
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/FelixKrueger/SNPsplit" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 496 512"><!--! Font Awesome Free 6.2.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"/></svg>
  </div>
  <div class="md-source__repository">
    FelixKrueger/SNPsplit
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Overview
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../workflow/" class="md-nav__link">
        SNPsplit workflow
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_3">
          Genome preparation
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Genome preparation" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Genome preparation
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          SNPsplit genome preparation
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        SNPsplit genome preparation
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#single-strain-mode" class="md-nav__link">
    Single strain mode:
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dual-strain-mode" class="md-nav__link">
    Dual strain mode:
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#filtering-high-confidence-snps-from-the-vcf-file" class="md-nav__link">
    Filtering high confidence SNPs from the VCF file
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#detecting-strains" class="md-nav__link">
    Detecting strains
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#detecting-chromosomes" class="md-nav__link">
    Detecting chromosomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dealing-with-variant-calls" class="md-nav__link">
    Dealing with Variant Calls
  </a>
  
    <nav class="md-nav" aria-label="Dealing with Variant Calls">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gt-genotype" class="md-nav__link">
    GT (Genotype)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fi-filter" class="md-nav__link">
    FI (Filter)
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#clearly-defined-clean-genotype" class="md-nav__link">
    Clearly defined clean genotype
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../legacy/" class="md-nav__link">
        Legacy (v5/GRCm38)
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" >
      
      
      
      
        <label class="md-nav__link" for="__nav_4">
          SNPsplit
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="SNPsplit" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          SNPsplit
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../SNPsplit/details/" class="md-nav__link">
        The SNPsplit workflow in more detail
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../SNPsplit/specific_comments/" class="md-nav__link">
        Specific comments
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Examples
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Examples" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Examples
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../Examples/examples/" class="md-nav__link">
        Examples
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6" type="checkbox" id="__nav_6" >
      
      
      
      
        <label class="md-nav__link" for="__nav_6">
          Full list of options
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Full list of options" data-md-level="1">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Full list of options
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../options/genome_prep_options/" class="md-nav__link">
        Genome preparation
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../options/SNPsplit_options/" class="md-nav__link">
        SNPsplit
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#single-strain-mode" class="md-nav__link">
    Single strain mode:
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dual-strain-mode" class="md-nav__link">
    Dual strain mode:
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#filtering-high-confidence-snps-from-the-vcf-file" class="md-nav__link">
    Filtering high confidence SNPs from the VCF file
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#detecting-strains" class="md-nav__link">
    Detecting strains
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#detecting-chromosomes" class="md-nav__link">
    Detecting chromosomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#dealing-with-variant-calls" class="md-nav__link">
    Dealing with Variant Calls
  </a>
  
    <nav class="md-nav" aria-label="Dealing with Variant Calls">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gt-genotype" class="md-nav__link">
    GT (Genotype)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#fi-filter" class="md-nav__link">
    FI (Filter)
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#clearly-defined-clean-genotype" class="md-nav__link">
    Clearly defined clean genotype
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  



<h1 id="snpsplit-genome-preparation">SNPsplit genome preparation</h1>
<p><code>SNPsplit_genome_preparation</code> is designed to read in a variant call file from the Mouse Genomes Project (this <a href="https://ftp.ebi.ac.uk/pub/databases/mousegenomes/REL-2112-v8-SNPs_Indels/mgp_REL2021_snps.vcf.gz">latest v8 file</a>) and generate new genome versions where the strain SNPs are either incorporated into the new genome (full sequence) or masked by the ambiguity nucleobase <code>N</code> (<strong>N-masking</strong>).</p>
<p><code>SNPsplit_genome_preparation</code> may be run in two different modes:</p>
<h3 id="single-strain-mode">Single strain mode:</h3>
<p><strong>1)</strong> The VCF file is read and filtered for high-confidence SNPs in the strain specified with <code>--strain &lt;name&gt;</code></p>
<p><strong>2)</strong> The reference genome (given with <code>--reference_genome &lt;genome&gt;</code>) is read into memory, and the filtered high-confidence SNP positions are incorporated either as N-masking (default), or full sequence (option <code>--full_sequence</code>)</p>
<h3 id="dual-strain-mode">Dual strain mode:</h3>
<p><strong>1)</strong> The VCF file is read and filtered for high-confidence SNPs in the strain specified with <code>--strain &lt;name&gt;</code></p>
<p><strong>2)</strong> The reference genome (given with <code>--reference_genome &lt;genome&gt;</code>) is read into memory, and the filtered high-confidence SNP positions are incorporated as full sequence and optionally as N-masking</p>
<p><strong>3)</strong> The VCF file is read one more time and filtered for high-confidence SNPs in strain 2 specified with <code>--strain2 &lt;name&gt;</code></p>
<p><strong>4)</strong> The filtered high-confidence SNP positions of strain 2 are incorporated as full sequence, and optionally as N-masking</p>
<p><strong>5)</strong> The SNP information of strain and strain 2 relative to the reference genome build are compared, and a new Ref/SNP annotation is constructed whereby the new Ref/SNP information will be Strain/Strain2 (and no longer the standard reference genome strain Black6 (C57BL/6J)). Later on, genome 1 will correspond to 'strain', genome 2 to 'strain2'.</p>
<p><strong>6)</strong> The full genome sequence given with <code>--strain &lt;name&gt;</code> is read into memory, and the high-confidence SNP positions between Strain and Strain2 are incorporated as full sequence, and optionally as N-masking</p>
<p>The resulting <code>.fa</code> files are ready to be indexed with your favourite aligner. Proven and tested aligners include Bowtie2, Tophat, STAR, HISAT2, HiCUP and Bismark. Please note that BWA does not support alignments to N-masked genome (see <a href="../../SNPsplit/specific_comments/#bwa">here</a>).</p>
<p>Both the SNP filtering and the genome preparation write out report files for record keeping.</p>
<h2 id="filtering-high-confidence-snps-from-the-vcf-file">Filtering high confidence SNPs from the VCF file</h2>
<p>This section describes in more detail the process of how high confidence SNPs are extracted from the VCF file <strong>mgp_REL2021_snps.vcf.gz</strong>. We are first going to paste the start of the VCF file and explain in more detail later the individual steps taken by the SNPsplit genome preparation for the strain CAST_EiJ as an example; the PDF version of the user guide had relevant information in the header lines or the variant data itself are marked in dark red, but unfotunately this is not supported in this markdown version, so please look carefully... . This should help you adapt the process to other genomes/VCF files should you wish to do so.</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>##fileformat=VCFv4.2
<a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>##FILTER=&lt;ID=PASS,Description=&quot;All filters passed&quot;&gt;
<a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a>##bcftoolsVersion=1.13+htslib-1.13
<a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a>##bcftoolsCommand=mpileup -f Mus_musculus.GRCm39.dna.toplevel.fa.gz -b samples -g 10 -a FORMAT/DP,FORMAT/AD,FORMAT/ADF,FORMAT/ADR,FORMAT/SP,INFO/AD -E -Q 0 -pm 3 -F 0.25 -d 500
<a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a>##reference=Mus_musculus.GRCm39.dna.toplevel.fa.gz
<a id="__codelineno-0-6" name="__codelineno-0-6" href="#__codelineno-0-6"></a>##contig=&lt;ID=1,length=195154279&gt;
<a id="__codelineno-0-7" name="__codelineno-0-7" href="#__codelineno-0-7"></a>##contig=&lt;ID=2,length=181755017&gt;
<a id="__codelineno-0-8" name="__codelineno-0-8" href="#__codelineno-0-8"></a>##contig=&lt;ID=3,length=159745316&gt;
<a id="__codelineno-0-9" name="__codelineno-0-9" href="#__codelineno-0-9"></a>##contig=&lt;ID=4,length=156860686&gt;
<a id="__codelineno-0-10" name="__codelineno-0-10" href="#__codelineno-0-10"></a>##contig=&lt;ID=5,length=151758149&gt;
<a id="__codelineno-0-11" name="__codelineno-0-11" href="#__codelineno-0-11"></a>##contig=&lt;ID=6,length=149588044&gt;
<a id="__codelineno-0-12" name="__codelineno-0-12" href="#__codelineno-0-12"></a>##contig=&lt;ID=7,length=144995196&gt;
<a id="__codelineno-0-13" name="__codelineno-0-13" href="#__codelineno-0-13"></a>##contig=&lt;ID=8,length=130127694&gt;
<a id="__codelineno-0-14" name="__codelineno-0-14" href="#__codelineno-0-14"></a>##contig=&lt;ID=9,length=124359700&gt;
<a id="__codelineno-0-15" name="__codelineno-0-15" href="#__codelineno-0-15"></a>##contig=&lt;ID=10,length=130530862&gt;
<a id="__codelineno-0-16" name="__codelineno-0-16" href="#__codelineno-0-16"></a>##contig=&lt;ID=11,length=121973369&gt;
<a id="__codelineno-0-17" name="__codelineno-0-17" href="#__codelineno-0-17"></a>##contig=&lt;ID=12,length=120092757&gt;
<a id="__codelineno-0-18" name="__codelineno-0-18" href="#__codelineno-0-18"></a>##contig=&lt;ID=13,length=120883175&gt;
<a id="__codelineno-0-19" name="__codelineno-0-19" href="#__codelineno-0-19"></a>##contig=&lt;ID=14,length=125139656&gt;
<a id="__codelineno-0-20" name="__codelineno-0-20" href="#__codelineno-0-20"></a>##contig=&lt;ID=15,length=104073951&gt;
<a id="__codelineno-0-21" name="__codelineno-0-21" href="#__codelineno-0-21"></a>##contig=&lt;ID=16,length=98008968&gt;
<a id="__codelineno-0-22" name="__codelineno-0-22" href="#__codelineno-0-22"></a>##contig=&lt;ID=17,length=95294699&gt;
<a id="__codelineno-0-23" name="__codelineno-0-23" href="#__codelineno-0-23"></a>##contig=&lt;ID=18,length=90720763&gt;
<a id="__codelineno-0-24" name="__codelineno-0-24" href="#__codelineno-0-24"></a>##contig=&lt;ID=19,length=61420004&gt;
<a id="__codelineno-0-25" name="__codelineno-0-25" href="#__codelineno-0-25"></a>##contig=&lt;ID=X,length=169476592&gt;
<a id="__codelineno-0-26" name="__codelineno-0-26" href="#__codelineno-0-26"></a>##ALT=&lt;ID=*,Description=&quot;Represents allele(s) other than observed.&quot;&gt;
<a id="__codelineno-0-27" name="__codelineno-0-27" href="#__codelineno-0-27"></a>##INFO=&lt;ID=INDEL,Number=0,Type=Flag,Description=&quot;Indicates that the variant is an INDEL.&quot;&gt;
<a id="__codelineno-0-28" name="__codelineno-0-28" href="#__codelineno-0-28"></a>##INFO=&lt;ID=IDV,Number=1,Type=Integer,Description=&quot;Maximum number of raw reads supporting an indel&quot;&gt;
<a id="__codelineno-0-29" name="__codelineno-0-29" href="#__codelineno-0-29"></a>##INFO=&lt;ID=IMF,Number=1,Type=Float,Description=&quot;Maximum fraction of raw reads supporting an indel&quot;&gt;
<a id="__codelineno-0-30" name="__codelineno-0-30" href="#__codelineno-0-30"></a>##INFO=&lt;ID=DP,Number=1,Type=Integer,Description=&quot;Raw read depth&quot;&gt;
<a id="__codelineno-0-31" name="__codelineno-0-31" href="#__codelineno-0-31"></a>##FORMAT=&lt;ID=PL,Number=G,Type=Integer,Description=&quot;List of Phred-scaled genotype likelihoods&quot;&gt;
<a id="__codelineno-0-32" name="__codelineno-0-32" href="#__codelineno-0-32"></a>##FORMAT=&lt;ID=DP,Number=1,Type=Integer,Description=&quot;Number of high-quality bases&quot;&gt;
<a id="__codelineno-0-33" name="__codelineno-0-33" href="#__codelineno-0-33"></a>##FORMAT=&lt;ID=AD,Number=R,Type=Integer,Description=&quot;Allelic depths (high-quality bases)&quot;&gt;
<a id="__codelineno-0-34" name="__codelineno-0-34" href="#__codelineno-0-34"></a>##INFO=&lt;ID=AD,Number=R,Type=Integer,Description=&quot;Total allelic depths (high-quality bases)&quot;&gt;
<a id="__codelineno-0-35" name="__codelineno-0-35" href="#__codelineno-0-35"></a>##INFO=&lt;ID=END,Number=1,Type=Integer,Description=&quot;End position of the variant described in this record&quot;&gt;
<a id="__codelineno-0-36" name="__codelineno-0-36" href="#__codelineno-0-36"></a>##INFO=&lt;ID=MinDP,Number=1,Type=Integer,Description=&quot;Minimum per-sample depth in this gVCF block&quot;&gt;
<a id="__codelineno-0-37" name="__codelineno-0-37" href="#__codelineno-0-37"></a>##FORMAT=&lt;ID=GT,Number=1,Type=String,Description=&quot;Genotype&quot;&gt;
<a id="__codelineno-0-38" name="__codelineno-0-38" href="#__codelineno-0-38"></a>##FORMAT=&lt;ID=GQ,Number=1,Type=Integer,Description=&quot;Phred-scaled Genotype Quality&quot;&gt;
<a id="__codelineno-0-39" name="__codelineno-0-39" href="#__codelineno-0-39"></a>##INFO=&lt;ID=DP4,Number=4,Type=Integer,Description=&quot;Number of high-quality ref-forward , ref-reverse, alt-forward and alt-reverse bases&quot;&gt;
<a id="__codelineno-0-40" name="__codelineno-0-40" href="#__codelineno-0-40"></a>##INFO=&lt;ID=MQ,Number=1,Type=Integer,Description=&quot;Average mapping quality&quot;&gt;
<a id="__codelineno-0-41" name="__codelineno-0-41" href="#__codelineno-0-41"></a>##bcftools_callCommand=call -mAv -f GQ,GP -p 0.99; Date=Wed Aug 11 21:20:03 2021
<a id="__codelineno-0-42" name="__codelineno-0-42" href="#__codelineno-0-42"></a>##bcftools_normCommand=norm --fasta-ref Mus_musculus.GRCm39.dna.toplevel.fa.gz -m +indels; Date=Fri Aug 13 11:11:49 2021
<a id="__codelineno-0-43" name="__codelineno-0-43" href="#__codelineno-0-43"></a>##FORMAT=&lt;ID=FI,Number=1,Type=Integer,Description=&quot;High confidence (1) or low confidence (0) based on soft filtering values&quot;&gt;
<a id="__codelineno-0-44" name="__codelineno-0-44" href="#__codelineno-0-44"></a>##FILTER=&lt;ID=LowQual,Description=&quot;Low quality variants&quot;&gt;
<a id="__codelineno-0-45" name="__codelineno-0-45" href="#__codelineno-0-45"></a>##VEP=&quot;v104&quot; time=&quot;2021-08-30 23:27:00&quot; cache=&quot;mus_musculus/104_GRCm39&quot; ensembl-funcgen=104.59ae779 ensembl-variation=104.6154f8b ensembl=104.1af1dce ensembl-io=104.1d3bb6e assembly=&quot;GRCm39&quot; dbSNP=&quot;150&quot; gencode=&quot;GENCODE M27&quot; regbuild=&quot;1
<a id="__codelineno-0-46" name="__codelineno-0-46" href="#__codelineno-0-46"></a>.0&quot; sift=&quot;sift&quot;
<a id="__codelineno-0-47" name="__codelineno-0-47" href="#__codelineno-0-47"></a>##INFO=&lt;ID=CSQ,Number=.,Type=String,Description=&quot;Consequence annotations from Ensembl VEP. Format: Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_position|Protein_position|Am
<a id="__codelineno-0-48" name="__codelineno-0-48" href="#__codelineno-0-48"></a>ino_acids|Codons|Existing_variation|DISTANCE|STRAND|FLAGS|VARIANT_CLASS|SYMBOL_SOURCE|HGNC_ID|SIFT|MOTIF_NAME|MOTIF_POS|HIGH_INF_POS|MOTIF_SCORE_CHANGE|TRANSCRIPTION_FACTORS&quot;&gt;
<a id="__codelineno-0-49" name="__codelineno-0-49" href="#__codelineno-0-49"></a>##bcftools_viewVersion=1.13+htslib-1.13
<a id="__codelineno-0-50" name="__codelineno-0-50" href="#__codelineno-0-50"></a>##bcftools_viewCommand=view -i &#39;FORMAT/FI[*] = 1&#39; mgp_REL2021_snps.vcf.gz; Date=Sat Dec 18 19:08:09 2021
<a id="__codelineno-0-51" name="__codelineno-0-51" href="#__codelineno-0-51"></a>##bcftools_annotateVersion=1.13+htslib-1.13
<a id="__codelineno-0-52" name="__codelineno-0-52" href="#__codelineno-0-52"></a>##bcftools_annotateCommand=annotate -x INFO/VDB,INFO/SGB,INFO/RPBZ,INFO/MQBZ,INFO/MQBZ,INFO/MQSBZ,INFO/BQBZ,INFO/SCBZ,INFO/FS,INFO/MQOF,INFO/AC,INFO/AN,FORMAT/SP,FORMAT/ADF,FORMAT/ADR,FORMAT/GP; Date=Sat Dec 18 19:08:09 2021
<a id="__codelineno-0-53" name="__codelineno-0-53" href="#__codelineno-0-53"></a>##INFO=&lt;ID=AC,Number=A,Type=Integer,Description=&quot;Allele count in genotypes&quot;&gt;
<a id="__codelineno-0-54" name="__codelineno-0-54" href="#__codelineno-0-54"></a>##INFO=&lt;ID=AN,Number=1,Type=Integer,Description=&quot;Total number of alleles in called genotypes&quot;&gt;
<a id="__codelineno-0-55" name="__codelineno-0-55" href="#__codelineno-0-55"></a>##bcftools_viewCommand=view -a -Oz -o final_mgp_REL2021_snps.vcf.gz; Date=Sat Dec 18 19:08:09 2021
<a id="__codelineno-0-56" name="__codelineno-0-56" href="#__codelineno-0-56"></a>##bcftools_annotateCommand=annotate -x INFO/MQ0F -Oz -o final_mgp_REL2021_snps.vcf.gz mgp_REL2021_snps.vcf.gz; Date=Mon Dec 20 07:12:23 2021
<a id="__codelineno-0-57" name="__codelineno-0-57" href="#__codelineno-0-57"></a>#CHROM  POS ID  REF ALT QUAL    FILTER  INFO    FORMAT  129P2_OlaHsd    129S1_SvImJ 129S5SvEvBrd    A_J AKR_J   B10.RIII    BALB_cByJ   BALB_cJ BTBR_T+_Itpr3tf_J   BUB_BnJ C3H_HeH C3H_HeJ C57BL_10J   C57BL_10SnJ C57BL_6NJ   C57BR_cdJ   C57L_J  C58_J   CAST_EiJ    CBA_J   CE_J    CZECHII_EiJ DBA_1J  DBA_2J  FVB_NJ  I_LnJ   JF1_MsJ KK_HiJ  LEWES_EiJ   LG_J    LP_J    MAMy_J  MOLF_EiJ    NOD_ShiLtJ  NON_LtJ NZB_B1NJ    NZO_HlLtJ   NZW_LacJ    PL_J    PWK_PhJ QSi3    QSi5    RF_J    RIIIS_J SEA_GnJ SJL_J   SM_J    SPRET_EiJ   ST_bJ   SWR_J   WSB_EiJ ZALENDE_EiJ
<a id="__codelineno-0-58" name="__codelineno-0-58" href="#__codelineno-0-58"></a>1   3050050 .   C   G   186.728 PASS    DP=3533;AD=3488,42;DP4=3041,447,44,1;MQ=60;CSQ=G|intergenic_variant|MODIFIER|||||||||||||||||||SNV||||||||,A|intergenic_variant|MODIFIER|||||||||||||||||||SNV||||||||;AC=2;AN=104  GT:PL:DP:AD:GQ:FI   0/0:0,57,168:19:19,0:71:1   0/0:0,108,255:36:36,0:122:1 0/0:0,30,154:10:10,0:44:1   0/0:0,199,255:66:66,0:127:1 0/0:0,84,222:28:28,0:98:1   0/0:0,208,255:69:69,0:127:1 0/0:0,163,255:54:54,
<a id="__codelineno-0-59" name="__codelineno-0-59" href="#__codelineno-0-59"></a>0:127:1 0/0:0,223,255:74:74,0:127:1 0/0:0,138,255:46:46,0:127:1 0/0:0,93,254:31:31,0:107:1  0/0:0,9,85:3:3,0:23:0   0/0:0,111,242:37:37,0:125:1 0/0:0,120,255:40:40,0:127:1 0/0:0,154,233:51:51,0:127:1 0/0:0,223,25
<a id="__codelineno-0-60" name="__codelineno-0-60" href="#__codelineno-0-60"></a>5:74:74,0:127:1 0/0:0,255,255:171:171,0:127:0   0/0:0,255,255:213:212,0:127:0   0/0:0,181,255:60:60,0:127:1 0/0:0,163,255:54:54,0:127:1 0/0:0,96,255:32:32,0:110:1  0/0:0,190,255:63:63,0:127:1 0/0:0,255,255:500:499,0:127:
<a id="__codelineno-0-61" name="__codelineno-0-61" href="#__codelineno-0-61"></a>0   0/0:0,93,255:31:31,0:107:1  0/0:0,96,255:32:32,0:110:1  0/0:0,51,233:17:17,0:65:1   0/0:0,90,255:30:30,0:104:1  0/0:0,255,255:220:220,0:127:0   0/0:0,255,255:106:106,0:127:1   0/0:0,196,255:65:65,0:127:0 0/0:
<a id="__codelineno-0-62" name="__codelineno-0-62" href="#__codelineno-0-62"></a>0,18,178:6:6,0:32:1 0/0:0,93,231:31:31,0:107:1  0/0:0,69,255:23:23,0:83:1   0/0:0,255,255:208:208,0:127:0   0/0:0,96,255:32:32,0:110:1  0/0:0,81,255:27:27,0:95:1   0/0:0,111,255:37:37,0:125:1 0/0:0,199,255:66:66,
<a id="__codelineno-0-63" name="__codelineno-0-63" href="#__codelineno-0-63"></a>0:127:1 0/0:0,144,255:49:48,0:127:1 0/0:0,57,255:19:19,0:71:1   0/0:0,255,255:366:366,0:127:0   0/0:0,93,255:31:31,0:107:1  0/0:0,78,255:26:26,0:92:1   0/0:0,69,255:23:23,0:83:1   0/0:0,187,255:62:62,0:127:1 0/0:
<a id="__codelineno-0-64" name="__codelineno-0-64" href="#__codelineno-0-64"></a>0,166,255:55:55,0:127:1 0/0:0,54,224:18:18,0:68:1   0/0:0,12,122:4:4,0:26:0 1/1:252,126,0:42:0,42:105:1 0/0:0,151,255:50:50,0:127:1 0/0:0,114,255:38:38,0:127:1 0/0:0,226,255:75:75,0:127:1 0/0:0,39,201:13:13,0:53:1
<a id="__codelineno-0-65" name="__codelineno-0-65" href="#__codelineno-0-65"></a>1   3050069 .   C   T   5422.31 PASS    DP=3674;AD=2868,796;DP4=2447,421,659,147;MQ=60;CSQ=T|intergenic_variant|MODIFIER|||||||||||||||||||SNV||||||||,A|intergenic_variant|MODIFIER|||||||||||||||||||SNV||||||||,G|interge
<a id="__codelineno-0-66" name="__codelineno-0-66" href="#__codelineno-0-66"></a>nic_variant|MODIFIER|||||||||||||||||||SNV||||||||;AC=45;AN=104 GT:PL:DP:AD:GQ:FI   1/1:198,72,0:24:0,24:67:1   1/1:255,120,0:40:0,40:115:1 1/1:149,27,0:9:0,9:22:1 0/0:0,208,255:69:69,0:127:1 1/1:253,90,0:35:1,34:85:1   0/0:0,226,255:75:75,0:127:1 0/0:0,166,255:55:55,0:127:1 0/0:0,199,255:66:66,0:127:1 1/1:255,160,0:53:0,53:127:1 1/1:255,96,0:32:0,32:91:1   0/0:0,12,119:4:4,0:10:0 0/0:0,129,255:43:43,0:127:1 0/0:0,123,255:41:41,
<a id="__codelineno-0-67" name="__codelineno-0-67" href="#__codelineno-0-67"></a>0:121:1 0/0:0,148,255:50:49,0:127:1 0/0:0,223,255:74:74,0:127:1 0/1:200,0,255:175:138,37:127:0  0/1:104,0,255:235:190,44:105:0  0/0:0,163,255:54:54,0:127:1 0/1:179,0,226:55:28,27:127:0    0/0:0,93,255:31:31,0:91:1   0/1:
<a id="__codelineno-0-68" name="__codelineno-0-68" href="#__codelineno-0-68"></a>255,0,255:65:35,30:127:0    0/0:0,255,255:493:467,26:127:0  0/0:0,96,255:32:32,0:94:1   0/0:0,105,255:35:35,0:103:1 1/1:234,72,0:24:0,24:67:1   1/1:255,90,0:30:0,30:85:1   0/0:0,255,255:228:219,8:127:0   0/0:0,255,25
<a id="__codelineno-0-69" name="__codelineno-0-69" href="#__codelineno-0-69"></a>5:109:109,0:127:1   0/0:0,181,255:61:60,0:127:0 0/0:0,24,170:9:8,0:22:1 1/1:239,96,0:32:0,32:91:1   1/1:252,63,0:22:0,21:58:1   0/0:0,255,255:228:226,1:127:0   1/1:255,105,0:35:0,35:100:1 1/1:255,84,0:28:0,28:79:1   0/1:255,0,178:36:17,19:127:0    0/0:0,208,255:70:69,0:127:1 0/0:0,138,255:46:46,0:127:1 1/1:255,75,0:25:0,25:70:1   0/0:0,255,255:387:368,18:127:0  1/1:255,123,0:41:0,41:118:1 1/1:255,87,0:29:0,29:82:1   0/0:0,81,255
<a id="__codelineno-0-70" name="__codelineno-0-70" href="#__codelineno-0-70"></a>:27:27,0:79:1   0/0:0,205,255:68:68,0:127:1 0/0:0,172,255:57:57,0:127:1 1/1:222,57,0:19:0,19:52:1   1/1:142,18,0:6:0,6:13:0 0/0:0,114,255:38:38,0:112:1 1/1:255,148,0:49:0,49:127:1 1/1:255,117,0:39:0,39:112:1 0/0:
<a id="__codelineno-0-71" name="__codelineno-0-71" href="#__codelineno-0-71"></a>0,208,255:70:69,0:127:1 1/1:251,48,0:16:0,16:43:1
<a id="__codelineno-0-72" name="__codelineno-0-72" href="#__codelineno-0-72"></a>…
</code></pre></div>
<h2 id="detecting-strains">Detecting strains</h2>
<p>To detect the available strains in the VCF file as well as to determine the column number of a desired strain in the file we skim through the header lines until we find a line starting with #CHROM. In terms of the strains in the file the next eight fields are irrelevant: </p>
<p><code>#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT</code></p>
<p>The following fields should contain the strains listed in the file, here:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>129S1_SvImJ 129S5SvEvBrd    A_J AKR_J   B10.RIII    BALB_cByJ   BALB_cJ BTBR_T+_Itpr3tf_J   BUB_BnJ C3H_HeH C3H_HeJ C57BL_10J   C57BL_10SnJ C57BL_6NJ   C57BR_cdJ   C57L_J  C58_J   CAST_EiJ    CBA_J   CE_J    CZECHII_EiJ DBA_1J  DBA_2J  FVB_NJ  I_LnJ   JF1_MsJ KK_HiJ  LEWES_EiJ   LG_J    LP_J    MAMy_J  MOLF_EiJ    NOD_ShiLtJ  NON_LtJ NZB_B1NJ    NZO_HlLtJ   NZW_LacJ    PL_J    PWK_PhJ QSi3    QSi5    RF_J    RIIIS_J SEA_GnJ SJL_J   SM_J    SPRET_EiJ   ST_bJ   SWR_J   WSB_EiJ ZALENDE_EiJ
</code></pre></div>
<p>In our example the data for strain <em>CAST_EiJ</em> would be found in column 28, or have an index of 27 (the index = column - 1 because index counting starts at 0 and not at 1). </p>
<p>If #CHROM is <strong>not present</strong> in the VCF file header, the <strong>automated lookup and subsequent steps will fail</strong>.</p>
<h2 id="detecting-chromosomes">Detecting chromosomes</h2>
<p>Chromosomes to be processed are detected from the VCF header lines starting with ##contig=<ID=…,… >, e.g. here:</p>
<p><div class="highlight"><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>##contig=&lt;ID=1,length=195154279&gt;
<a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>##contig=&lt;ID=2,length=181755017&gt;
<a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a>…
</code></pre></div>
The IDs here are the chromosomes for which variant calls are available, but not necessarily all chromosomes in the genome have to be present here. If there are no variant calls for say “chrY” then the subsequent genome preparation step will simply not introduce any changes but use the reference sequence for “chrY”.</p>
<p>If the ##contig=… fields are missing from the VCF file, <strong>subsequent steps are probably going to fail</strong>; as a way around this you might get away with defining the chromosome array manually, e.g. like so:</p>
<p><code>my @chroms = (1..19,’X’,’Y’,’MT’);</code></p>
<h2 id="dealing-with-variant-calls">Dealing with Variant Calls</h2>
<p>For variant calls the SNPsplit genome preparation extracts the following information from each line:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a>CHROM  [Col 1]
<a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>POS    [Col 2]
<a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a>REF    [Col 4]
<a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a>ALT    [Col 5]
<a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a>STRAIN [Col 28 (for CAST_EiJ)]
</code></pre></div>
<p>Which in our case is (three example lines only):</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a>CHROM   POS REF ALT CAST_EiJ
<a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a>1   3050050 C   G   1/1:0,163,255:54:54,0:127:0
<a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>1   3050076 C   T   0/0:0,214,255:71:71,0:127:0 
<a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a>1   3050069 C   T   1/1:234,72,0:24:0,24:67:1
</code></pre></div>
<p>Now the STRAIN field contains a lot of information which is specified in the FORMAT field and further ‘explained’ in the VCF header. The format is: 
<div class="highlight"><pre><span></span><code><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a>GT:PL:DP:AD:GQ:FI
</code></pre></div></p>
<p>I am not going into the details about all these FORMAT tags (feel free to browse the header section above), but suffice it to say that the SNPsplit genome preparation only cares about the GT (=GENOTYPE) and FI (=FILTER) entry which are defined as:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>##FORMAT=&lt;ID=GT,Number=1,Type=String,Description=&quot;Genotype&quot;&gt;
</code></pre></div>
<h3 id="gt-genotype">GT (Genotype)</h3>
<p>The GT string can be one of the following:
<div class="highlight"><pre><span></span><code><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a>&#39;.&#39;    = no genotype call was made
<a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a>&#39;0/0&#39;  = genotype is the same as the reference genome
<a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>&#39;1/1&#39;  = homozygous alternative allele; can also be &#39;2/2&#39;, &#39;3/3&#39;, etc. if more than one alternative allele is present.
<a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a>&#39;0/1&#39;  = heterozygous genotype; can also be &#39;1/2&#39;, &#39;0/2&#39;, etc.
</code></pre></div></p>
<p>For the mouse strains we are working with here we only accept homozygous alternative alleles, so '1/1', ‘2/2’ or ‘3/3’. Any other combination is recorded and mentioned in the final report but not included in the SNP file.</p>
<h3 id="fi-filter">FI (Filter)</h3>
<div class="highlight"><pre><span></span><code><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>##FORMAT=&lt;ID=FI,Number=1,Type=Integer,Description=&quot;High confidence (1) or low confidence (0) based on soft filtering values&quot;&gt;
</code></pre></div>
<p>Next we are looking at the FILTER field. If a SNP variant call passed all filters for a given strain it will PASS or get a value of 1. Else it will have a value of 0. In the example above the first position had a genotype of <strong>1/1</strong> but did not pass the Filter for the CAST_EiJ strain (FI value = 0), so this position won’t be included:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a>1   3050050 C   G   1/1:0,163,255:54:54,0:127:0 
</code></pre></div>
<p>The second position did not indicate that there was a SNP (<strong>0/0</strong>), so the positions wouldn’t be considered for a single-hybrid genome anyway. However, the filter call for that position was uncertain (FI value = 0), so this position would also not be considered for a SNP position in <code>--dual_hybrid</code> mode even if the other strain had a high confidence SNP at this position (this behaviour was introduced in v0.3.2).</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a>1   3050076 C   T   0/0:0,214,255:71:71,0:127:0
</code></pre></div>
<p>The third position finally is a high-confidence homozygous SNP which would be included for both single and dual hybrid genomes:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a>1   3050069 C   T   1/1:234,72,0:24:0,24:67:1
</code></pre></div>
<h2 id="clearly-defined-clean-genotype">Clearly defined clean genotype</h2>
<p>And finally, variants are required to have a clearly defined genotype, e.g. a made-up position:
<div class="highlight"><pre><span></span><code><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a>12  45630185    A   T/C
</code></pre></div></p>
<p>would not be included as a valid SNP position because the alternative allele is not clearly defined (unless the genotype would be 2/2 here, in which case the reference would be ‘A’ and the alternative allele would be ‘C’).</p>
<p>If you are dealing with any VCF file other than the one used as an example here you might need to adapt one or more of the issues addressed here to make it work.</p>





                
              </article>
            </div>
          
          
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.instant", "navigation.tabs"], "search": "../../assets/javascripts/workers/search.12658920.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.5cf534bf.min.js"></script>
      
    
  </body>
</html>